src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -378,11 +378,23 @@ public TypeInfoWithDebugInformation(TypeInfo typeInfo, int debugId, string name)
     internal class MonoBinaryReader : BinaryReader
     {
         public bool HasError { get; }
-        public MonoBinaryReader(Stream stream, bool hasError = false) : base(stream)
+
+        private MonoBinaryReader(Stream stream, bool hasError = false) : base(stream)
         {
             HasError = hasError;
         }
 
+        public MonoBinaryReader(byte [] data) : base(new MemoryStream(data)) {}
+
+        public static MonoBinaryReader From(Result result)
+        {
+            byte[] newBytes = Array.Empty<byte>();
+            if (!result.IsErr) {
+                newBytes = Convert.FromBase64String(result.Value?["result"]?["value"]?["value"]?.Value<string>());
```

Comment:
This can NRE

Reply:
The code has just moved, it is the same as before.